paritytech/polkassembly
# Add and implement comment of comments

- closes #21
- quite some cleanup creating ui-components and components for comment**s**, comment, creation label, update label..
- renaming reply to comment
- arbitrary comment depth to 3

How to test:
- Migrate hasura (or use the `launch.sh` at the root)
- Create or make sure you have at least one entry in `post_type` and `post_topic` tables.
- Create a post
- Create a comment or 2. Comments with `null` as `parent_comment_id` are considered as root comments. if you specify a comment id, then it'll be a nested comment.

You can then do queries such as:
```gql
query MyQuery {
  __typename
  posts {
    author{
      username
      id
    }
    id
    comments(where: {parent_comment_id: {_is_null: true}}) {
      author {
        id
        username
      }
      content
      comments {
        author {
          id
          username
        }
        content
        comments {
        author {
          id
          username
        }
        content
      }
      }
    }
  }
}
```

That's how it looks like:
![image](https://user-images.githubusercontent.com/33178835/70926082-be34c280-202c-11ea-945a-6d6eca130188.png)
